src/Sound/Tidal/Stream/Process.hs

```diff
@@ -140,7 +140,7 @@ processCps cconf cref ss = mapM processEvent
       onPart <- Clock.timeAtBeat cconf ss partStartBeat
       when (eventHasOnset e) (do
         let cps' = Map.lookup "cps" (value e) >>= getF
-        maybe (return ()) (\newCps -> Clock.setCPS cconf cref newCps) (fmap toRational cps')
+        maybe (return ()) (\newCps -> Clock.setTempoCPS newCps on cconf ss) (fmap toRational cps')
```

Comment:
Doesn't this mean we will lose the tempo changes that happen in the single tick case since we don't (and can't) commit the zeroed session state?

Reply:
i thought so too, but `streamOnce tidal $ cps 1` is still working.. not completely sure why?